bitcoin/src/blockdata/witness.rs

```diff
@@ -95,16 +108,37 @@ impl Decodable for Witness {
                 cursor += element_size;
             }
             content.truncate(cursor);
+            // Index space is now at the end of the Vec
+            content.rotate_left(witness_index_space);
             Ok(Witness {
                 content,
                 witness_elements,
-                last,
-                second_to_last,
+                indices_start: cursor - witness_index_space,
             })
         }
     }
 }
 
+
+/// Safety Requirements: value must always fit within u32
+#[inline]
+fn encode_cursor(bytes: &mut [u8], start_of_indices: usize, index: usize, value: usize) {
+    let start = start_of_indices + index * U32_SIZE;
+    let end = start + U32_SIZE;
+    bytes[start..end].copy_from_slice(&(value as u32).to_ne_bytes()[..]);
```

Comment:
In 4226d602057529885cf95f935370f6e5bf68bfd2:

I think we should use `.try_into().unwrap()` (or `expect`) rather than a cast.